cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -174,22 +181,53 @@ async def _batch_async(
 
         return await asyncio.wait_for(asyncio.gather(*tlm_query_tasks), timeout=timeout)  # type: ignore[arg-type]
 
-    def prompt(self, prompt: str, options: Optional[TLMOptions] = None) -> TLMResponse:
+    def prompt(
+        self,
+        prompt: Union[str, Collection[str]],
+        options: Union[None, TLMOptions, Collection[Union[TLMOptions, None]]] = None,
+        timeout: Optional[float] = None,
+        retries: int = 1,
+    ) -> Union[TLMResponse, List[TLMResponse]]:
         """
         Get response and confidence from TLM.
 
         Args:
-            prompt (str): prompt for the TLM
-            options (Optional[TLMOptions]): options to parameterize TLM with. Defaults to None.
+            prompt (str | Collection[str]): prompt (or list/iterable of multiple prompts) for the TLM
+            options (None | TLMOptions | Collection[TLMOptions |  None], optional): collection of options (or instance of options) to pass to prompt method. Defaults to None.
+            timeout (Optional[float], optional): timeout (in seconds) to run all prompts. Defaults to None.
+                If the timeout is hit, this method will throw a `TimeoutError`.
+                Larger values give TLM a higher chance to return outputs for all of your prompts.
+                Smaller values ensure this method does not take too long.
+            retries (int): number of retries to attempt for each individual prompt in case of internal error. Defaults to 1.
+                Larger values give TLM a higher chance of returning outputs for all of your prompts,
+                but this method will also take longer to alert you in cases of an unrecoverable error.
+                Set to 0 to never attempt any retries.
         Returns:
-            TLMResponse: [TLMResponse](#class-tlmresponse) object containing the response and confidence score
+            TLMResponse | List[TLMResponse]: [TLMResponse](#class-tlmresponse) object containing the response and confidence score.
+                    If multiple prompts were provided in a list, then a list of such objects is returned, one for each prompt.
         """
-        return self._event_loop.run_until_complete(
-            self.prompt_async(
+        if is_collection(prompt):
+            return self._batch_prompt(
                 prompt,
                 options,
+                timeout=timeout,
+                retries=retries,
             )
-        )
+
+        elif isinstance(prompt, str):
+            if not isinstance(options, dict):
+                raise ValueError("options must be a single TLMOptions object for single prompt.")
```

Comment:
[nitnit, maybe next PR] User would not know what "TLMOptions" object is without diving deep into code. Maybe here we can say a single dictionary for clarity?